tools/depcheck/pkg/cmd/pin.go

```diff
@@ -0,0 +1,131 @@
+	YamlFile *glide.YamlFile
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type PinImportsFlags struct {
+	LockFileName string
+	YamlFileName string
+}
+
+func (o *PinImportsFlags) Run(out, errout io.Writer) (*PinImportsOpts, error) {
+	if len(o.LockFileName) == 0 || len(o.YamlFileName) == 0 {
+		return nil, fmt.Errorf("both a glide.lock file and a glide.yaml file must be specified")
+	}
+
+	lockFileBytes, err := ioutil.ReadFile(o.LockFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	yamlFileBytes, err := ioutil.ReadFile(o.YamlFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	lockfile := &glide.LockFile{}
+	yamlfile := &glide.YamlFile{}
+
+	if err := lockfile.Decode(lockFileBytes); err != nil {
+		return nil, err
+	}
+
+	if err := yamlfile.Decode(yamlFileBytes); err != nil {
+		return nil, err
+	}
+
+	return &PinImportsOpts{
+		LockFile: lockfile,
+		YamlFile: yamlfile,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdPinImports(parent string, out, errout io.Writer) *cobra.Command {
+	flags := &PinImportsFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "pin --glide.lock=FOO.lock --glide.yaml=FOO.yaml",
+		Short: "Updates all non-forked dependencies in a glide.yaml file to use the SHAs specified in a glide.lock file.",
+		Long: `Updates all non-forked dependencies in a glide.yaml file to use the SHAs specified in a glide.lock file.
+Any existing dependencies on a glide.yaml file containing a "repo" field are ignored.`,
+		Example: fmt.Sprintf(pinImportsExample, parent),
+		Run: func(c *cobra.Command, args []string) {
+			opts, err := flags.Run(out, errout)
+			if err != nil {
+				fmt.Fprintf(errout, "%v\n", err)
+				os.Exit(1)
+			}
+
+			if err := opts.Validate(); err != nil {
+				fmt.Fprintf(errout, "%v\n", err)
+				os.Exit(1)
+			}
+			if err := opts.Run(); err != nil {
+				fmt.Fprintf(errout, "%v\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&flags.LockFileName, "glide.lock", "glide.lock", "Path to glide.lock file used to indicate which dependencies should be updated or pinned to a specific revision in the glide.yaml file")
+	cmd.Flags().StringVar(&flags.YamlFileName, "glide.yaml", "glide.yaml", "Path to glide.yaml file used to pin dependencies from the given --glide.lock file")
+
+	return cmd
+}
+
+func (o *PinImportsOpts) Validate() error {
+	if o.YamlFile == nil || o.LockFile == nil {
+		return fmt.Errorf("both a glide.yaml and a glide.lock file are required")
+	}
+	if o.Out == nil || o.ErrOut == nil {
+		return fmt.Errorf("no output or error output buffer given")
+	}
+
+	return nil
+}
+
+func (o *PinImportsOpts) Run() error {
+	missing, err, warnings := glide.MissingImports(o.LockFile, o.YamlFile)
+	if err != nil {
+		return fmt.Errorf("error: unable to calculate missing imports: %v\n", err)
+	}
+
+	for _, w := range warnings {
+		fmt.Fprintf(o.ErrOut, w)
+	}
+
+	missingBytes, err := missing.Encode()
+	if err != nil {
+		return err
+	}
+
+	o.Out.Write(o.YamlFile.DecodedBytes)
```

Comment:
Put the bytes on your options struct.  It's weird to keep track of your decoded bytes.  They won't match your object later on.